pkgs/games/am2r-launcher/default.nix

```diff
@@ -0,0 +1,105 @@
+{ lib
+, buildDotnetModule
+, pkgsi686Linux
+, writeShellScript
+, glibc
+, gtk3
+, libappindicator
+, webkitgtk
+, e2fsprogs
+, libnotify
+, libgit2
+, openssl
+, xdelta
+, file
+, busybox
+, openjdk
+, patchelf
+, fetchFromGitHub
+, glib-networking
+}:
+let
+  am2r-run = pkgsi686Linux.buildFHSEnv {
```

Comment:
as someone not as familiar with nix, what does this wrapper do?

Reply:
Essentially, it creates an isolated sandbox that follows the Linux Filesystem Hierarchy Standard. In Nix, all packages live in `/nix/store/` rather than in `/lib`, `/bin`, `/usr`, etc. When running a binary that expects that sort of environment and it isn't feasible to recompile or patch via `patchelf`, `buildFHSEnv` allows Nix to construct an environment that looks like a more "normal" Linux system and includes every library/package that the binary needs. This is most extensively used for games or proprietary software, where the code is closed-source and the binary isn't really modifiable.

In this example, we're making a sandbox that contains all the 32-bit packages that the game binary needs to run. The `am2r-run-binary.patch` file also patches the launcher so that the `runner` file is launched with the `am2r-run` script. This allows the binary to run in the FHS environment and find all of its dependencies.

Reply:
Just a reminder, this needs to be updated to support the 64 bit AM2R 1.6, https://github.com/NixOS/nixpkgs/pull/253203#issuecomment-1801405605.

Reply:
Ah thanks for the explanation. 
I did see the patch, and was wondering whether it makes sense to somehow get that into upstream. But considering that the nix-store is a very nix concept, it feels nicer having that as a patch.